app/auth/callback/route.ts

```diff
@@ -30,7 +30,7 @@ export async function GET(request: Request) {
 
     const { error } = await supabase.auth.exchangeCodeForSession(code);
     if (!error) {
-      return NextResponse.redirect(`${origin}${next}`);
+      return NextResponse.redirect(`${origin}/${next}`);
```

Comment:
Same thing here as below, we should be using the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL_API). I know this is not your code originally, but its stuff like this that makes it easier to do routing logic idiomatically. Without need to manged trailing/leading slashes:

> See how all these have the same href:
```
Welcome to Node.js v22.3.0.
Type ".help" for more information.
> new URL("apply", "https://example.com").href
'https://example.com/apply'
> new URL("/apply", "https://example.com").href
'https://example.com/apply'
> new URL("/apply/", "https://example.com").href
'https://example.com/apply/'
> new URL("apply/", "https://example.com").href
'https://example.com/apply/'
```